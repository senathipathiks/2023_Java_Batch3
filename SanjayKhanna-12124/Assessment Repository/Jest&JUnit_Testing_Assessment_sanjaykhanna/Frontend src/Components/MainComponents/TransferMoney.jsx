import { Button, FormControl, InputLabel, MenuItem, Select, TextField } from '@mui/material';
import InputAdornment from '@mui/material/InputAdornment';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import Output from '../UtilitiesComponents/Output';
const services = require('../Services/Services')

const TransferMoney = () => {

    
    const [payeeData, setPayeeData] = useState([]);
    const [insertData, setInsertData] = useState({
        amount: '',
        payee: {
            payeeId: ''
        }
    })

    const [msg, setMsg] = useState("");
    const [rightFlag, setRightFlag] = useState()
    const [wrongFlag, setWrongFlag] = useState()


    useEffect(() => {
        
        services.doGetAllPayee().then((res)=>{
            return res.data;
        }).then((resp)=>{
            setPayeeData(resp)
        }).catch(err => {
            console.log(err)
        })
    },[])

    const handleSubmit = (e) =>{
        e.preventDefault();
        services.doSave(insertData).then(res=>{
            if (res.data === "success") {
                setMsg("Transaction Successful")
                setRightFlag(true)
            }
            else {
                setMsg("Transaction Failure!")
                setWrongFlag(true)
            }
        }).catch(err=>{
            console.log(err)
        })
        setTimeout(()=>{
            window.location.href="/"
        },2000)
    }

    return (
        <div>
            <div className='container w-25 p-5 shadow bg-body rounded position-absolute top-50 start-50 translate-middle'>
                <div className='mb-3 text-center h5' data-testid="transferHeader">Enter details to Transfer Money</div>
                <form onSubmit={(e)=>handleSubmit(e)}>
                    <FormControl fullWidth className="mt-3">
                        <InputLabel id="payment-label">Select Payee's Nick Name</InputLabel>
                        <Select
                            labelId="payment-label"
                            id='payeeId'
                            name='payeeId'
                            label="Select Payee's Nick Name"
                            aria-label='payeeId'
                            required
                            onChange={(e) => { setInsertData({ ...insertData, payee: { payeeId: e.target.value } }) }}
                        >
                            {payeeData && payeeData.map(item => <MenuItem key={item.payeeId} value={item.payeeId}>{item.nickName}</MenuItem>)}
                        </Select>
                    </FormControl>
                    <TextField
                        className="mt-3"
                        required
                        autoComplete='off'
                        id="amount"
                        name="amount"
                        label="amount"
                        aria-label='amount'
                        type="number"
                        error= {insertData.amount < 1 && insertData.amount !== "" ? true : false}
                        helperText={insertData.amount < 1 && insertData.amount !== "" ? "Amount should be greater than 0":""}
                        fullWidth
                        InputProps={{
                            startAdornment: <InputAdornment position="start">Rs:</InputAdornment>,
                        }}
                        variant="standard"
                        onChange={(e) => { setInsertData({ ...insertData, [e.target.name]: e.target.value }) }}
                    />
                    <div className='mt-3'>
                        <Link to="/" className="float-start">
                            <Button variant="contained" color="success" >
                                <i className="bi bi-arrow-left"></i>back
                            </Button>
                        </Link>

                        <Button type="submit"  disabled={insertData.amount <= 0 && insertData.amount !== "" ? true:false} className="float-end" variant="contained">
                            Transfer
                        </Button>
                    </div>
                </form>
            </div>
            {(rightFlag || wrongFlag) && <Output success={rightFlag} fail={wrongFlag} msg={msg} />}
        </div>
    );
}

export default TransferMoney;
