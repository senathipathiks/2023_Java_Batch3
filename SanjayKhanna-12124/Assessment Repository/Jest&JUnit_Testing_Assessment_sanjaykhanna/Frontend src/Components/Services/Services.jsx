
import axios from 'axios';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"

const doAdd = (data) => {
    return axios.post(origin + addPayee, data);
}

const doDelete = (payeeId) => {
    return axios.delete(origin + deletePayee + payeeId);
}

const doUpdate = (data) => {
    return axios.put(origin + updatePayee, data);
}

const doGetPayee = (payeeId) => {
    return axios.get(origin + getPayee + payeeId);
}

const doGetAllPayee = () => {
    return axios.get(origin + getPayees);
}

const doSave = (data) => {
    return axios.post(origin + saveTransaction, data);
}

const doGetAllTransaction = () => {
    return axios.get(origin + getTransactions);
}

// exports.doAdd = doAdd;
// exports.doDelete = doDelete;
// exports.doGetAllPayee = doGetAllPayee;
// exports.doGetAllTransaction = doGetAllTransaction;
// exports.doGetPayee = doGetPayee;
// exports.doSave = doSave;
// exports.doUpdate = doUpdate;


export { doAdd, doDelete, doGetAllPayee, doGetAllTransaction, doGetPayee, doSave, doUpdate };

