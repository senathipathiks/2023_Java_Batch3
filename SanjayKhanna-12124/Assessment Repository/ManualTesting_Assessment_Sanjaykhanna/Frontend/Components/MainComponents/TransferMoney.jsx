import { Button, FormControl, InputLabel, MenuItem, Select, TextField } from '@mui/material';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import payeeService from '../../Services/PayeeService';
import transactionService from '../../Services/TransactionService';
import Output from '../UtilitiesComponents/Output';

const TransferMoney = () => {

    const [payeeData, setPayeeData] = useState([]);
    const [insertData, setInsertData] = useState({
        amount: '',
        payee: {
            payeeId: ''
        }
    })

    const [msg, setMsg] = useState("");
    const [rightFlag, setRightFlag] = useState()
    const [wrongFlag, setWrongFlag] = useState()


    useEffect(() => {
        payeeService.doGetAll()
            .then((res) => {
                setPayeeData(res)
            }).catch(err => {
                console.log(err)
            })
    },[])

    const handleSubmit = (e) =>{
        e.preventDefault();
        transactionService.doSave(insertData).then(res=>{
            if (res.data === "success") {
                setMsg("Transaction Successful")
                setRightFlag(true)
            }
            else {
                setMsg("Transaction Failure!")
                setWrongFlag(true)
            }
        }).catch(err=>{
            console.log(err)
        })
        setTimeout(()=>{
            window.location.href="/"
        },5000)
    }

    return (
        <div>
            <div className='container w-25 p-5 shadow bg-body rounded position-absolute top-50 start-50 translate-middle'>
                <div className='mb-3 text-center h5'>Enter details to Transfer Money</div>
                <form onSubmit={(e)=>handleSubmit(e)}>
                    <FormControl fullWidth className="mt-3">
                        <InputLabel id="payment-label">Select Payee's Nick Name</InputLabel>
                        <Select
                            labelId="payment-label"
                            id='payeeId'
                            name='payeeId'
                            label="Choose Payment Method"
                            required
                            onChange={(e) => { setInsertData({ ...insertData, payee: { payeeId: e.target.value } }) }}
                        >
                            {payeeData!=null && payeeData.map(item => <MenuItem value={item.payeeId}>{item.nickName}</MenuItem>)}
                        </Select>
                    </FormControl>
                    <TextField
                        className="mt-3"
                        autoFocus
                        required
                        autoComplete='off'
                        id="amount"
                        name="amount"
                        label="amount"
                        type="number"
                        fullWidth
                        variant="standard"
                        onChange={(e) => { setInsertData({ ...insertData, [e.target.name]: e.target.value }) }}
                    />
                    <div className='mt-3'>
                        <Link to="/" className="float-start">
                            <Button variant="contained" color="success">
                                <i className="bi bi-arrow-left"></i>back
                            </Button>
                        </Link>

                        <Button type="submit" className="float-end" variant="contained">
                            Transfer
                        </Button>
                    </div>
                </form>
            </div>
            {(rightFlag || wrongFlag) && <Output success={rightFlag} fail={wrongFlag} msg={msg} />}
        </div>
    );
}

export default TransferMoney;
