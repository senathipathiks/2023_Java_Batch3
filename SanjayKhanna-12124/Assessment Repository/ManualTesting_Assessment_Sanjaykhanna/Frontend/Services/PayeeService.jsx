import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="

class PayeeService extends Component {
 
    doAdd = (data) =>{
        return axios.post(origin+addPayee,data);
    }

    doDelete = (payeeId) =>{
        return axios.delete(origin+deletePayee+payeeId);
    }

    doUpdate = (data) => {
        return axios.put(origin+updatePayee,data);
    }

    doGetPayee = (payeeId) =>{
        return axios.get(origin+getPayee+payeeId);
    }

    doGetAll = () => {
        return axios.get(origin+getPayees);
    }
}

const payeeService = new PayeeService();
export default payeeService;
