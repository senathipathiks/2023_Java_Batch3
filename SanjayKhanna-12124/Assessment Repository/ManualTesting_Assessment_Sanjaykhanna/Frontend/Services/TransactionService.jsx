import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"
class TransactionService extends Component {
  
    doSave = (data) => {
        return axios.post(origin+saveTransaction,data);
    }

    doGetAll = () =>{
        return axios.get(origin+getTransactions);
    }
}
const transactionService = new TransactionService();
export default transactionService;
