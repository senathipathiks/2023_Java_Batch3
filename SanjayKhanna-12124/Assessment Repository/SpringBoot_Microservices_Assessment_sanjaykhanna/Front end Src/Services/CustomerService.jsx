import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createCustomer = "createcustomer";
const readAllCustomer = "readallcustomer";
const updateCustomer = "updatecustomer";
const deleteCustomer = "deletecustomer?id=";
const readCustomer = "readcustomer?id=";

class CustomerService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createCustomer, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllCustomer);
    };

    doRead = (id) => {
        return axios.get(origin + readCustomer + id);
    };

    doUpdate = (data) => {
        return axios.put(origin + updateCustomer, data);
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteCustomer + id)
    };
}
const customerService = new CustomerService();
export default customerService;
