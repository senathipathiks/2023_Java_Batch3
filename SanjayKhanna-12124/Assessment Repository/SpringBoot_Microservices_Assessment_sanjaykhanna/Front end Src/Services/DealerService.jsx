import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createDealer = "createdealer";
const readAllDealer = "readalldealer";
const updateDealer = "updatedealer";
const deleteDealer = "deletedealer?id=";
const readDealer = "readdealer?id=";
class DealerService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createDealer, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllDealer);
    };

    doRead = (id) => {
        return axios.get(origin + readDealer + id);
    };

    doUpdate = (data) => {
        return axios.put(origin + updateDealer, data);
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteDealer + id)
    };

}
const  dealerService = new DealerService();
export  default dealerService;