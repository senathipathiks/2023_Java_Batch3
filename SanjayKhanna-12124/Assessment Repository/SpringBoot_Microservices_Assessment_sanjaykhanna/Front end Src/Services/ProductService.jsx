import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createProduct = "createproduct";
const readAllProduct = "readallproduct";
const updateProduct = "updateproduct";
const deleteProduct = "deleteproduct?id=";
const readProduct = "readproduct?id=";

class ProductService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createProduct, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllProduct);
    };

    doRead = (id) => {
        return axios.get(origin + readProduct + id);
    };

    doUpdate = (data) =>{
        return axios.put(origin+updateProduct, data); 
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteProduct + id)
    };

}
const  productService = new ProductService();
export default productService;
