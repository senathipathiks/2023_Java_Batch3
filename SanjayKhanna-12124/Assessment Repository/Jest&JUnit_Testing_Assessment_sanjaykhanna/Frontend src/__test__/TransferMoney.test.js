import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TransferMoney from '../Components/MainComponents/TransferMoney';
import { doSave } from '../Components/Services/Services';
jest.mock('react-router-dom')

describe('Checking the Transfer Money Component', () => {

    test('should render the Transfer Money Component', () => {
        render(<TransferMoney />)
        const transferMoney = screen.getByText('Enter details to Transfer Money');
        expect(transferMoney).toBeInTheDocument();
    });

    test('should render the Header of transfer money component', () => {
        render(<TransferMoney />)
        expect(screen.getByTestId('transferHeader')).toBeInTheDocument();
        expect(screen.getByTestId('transferHeader')).toHaveTextContent('Enter details to Transfer Money');
    });

    test('should check the field in transfer money component', () => {
        render(<TransferMoney />)
        expect(screen.getByLabelText("amount")).toBeInTheDocument();
        expect(screen.getByLabelText("payeeId")).toBeInTheDocument();
    });

    test('should perform the transfer method', () => {
        const data = {
            amount: 2000,
            payee: {
                payeeId: 10
            }
        }
        const result = doSave(data)
        result.then(res => {
                expect(res.status).toBe(200);
                expect(res.data).toEqual("success");
                })
            .catch(err => {
                console.log(err)
            })
    });


    test('should render the Transfer Money Component', () => {
        expect(renderer.create(<TransferMoney />).toJSON()).toMatchSnapshot();
    });
});
