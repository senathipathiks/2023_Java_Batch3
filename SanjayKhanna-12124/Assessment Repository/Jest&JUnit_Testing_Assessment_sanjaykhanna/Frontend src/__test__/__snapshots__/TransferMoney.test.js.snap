// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking the Transfer Money Component should render the Transfer Money Component 1`] = `
<div>
  <div
    className="container w-25 p-5 shadow bg-body rounded position-absolute top-50 start-50 translate-middle"
  >
    <div
      className="mb-3 text-center h5"
      data-testid="transferHeader"
    >
      Enter details to Transfer Money
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth mt-3 css-q8hpuo-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          id="payment-label"
        >
          Select Payee's Nick Name
        </label>
        <div
          aria-label="payeeId"
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
          required={true}
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="payment-label payeeId"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="payeeId"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden={true}
            aria-invalid={false}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="payeeId"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={true}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Select Payee's Nick Name
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        aria-label="amount"
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mt-3 css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          htmlFor="amount"
          id="amount-label"
        >
          amount
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-953pxc-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
            >
              Rs:
            </p>
          </div>
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="amount"
            name="amount"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="number"
          />
        </div>
      </div>
      <div
        className="mt-3"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary float-end css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Transfer
        </button>
      </div>
    </form>
  </div>
</div>
`;
